### DISCUSSION TITLE
Zero-one-inflated beta?

### DISCUSSION THREAD

**Discussion by maxlindmark** [sdmTMB/sdmTMB#229] [Ideas]
Hi!

During a sdmTMB workshop a question came up on how to model proportion data with many zeroes and ones. The example was proportions of herring in tows of an "unintentionally mixed" sprat and herring fishery (essentially bycatch modelling). Typical for the data would be something like 25% 0's, 25% 1's and rest in between*, so a delta-beta would not really help unless the 1's are modified with some constants.  

Would it be difficult/interesting/doable to implement the zero-one-inflated-beta regression, e.g., [as in brms](https://cran.rstudio.com/web/packages/brms/vignettes/brms_families.html#zero-inflated-and-hurdle-models)? Or do you have other suggestions on how to tackle this with sdmTMB (there are likely spatial trends in data)?

* edited: these proportions are just examples to illustrate that the 0's and 1's are not rare!

**Comment by ericward-noaa** [comment]
This is a cool suggestion. Currently there's a delta_beta() family implemented, and you're right -- this would not handle the proportions. That would need to have to be more of a ZOIB distribution -- this is not hard to implement for a case where covariates aren't affecting the contributions to the 3 part mixture (0, 1, everything else) -- but if covariates / spatial effects are thought to impact these differently, then it'll be more work. The formula argument and others are set up to be a list (for delta models) -- but there's a lot of internal hardwiring that is restricting the dimension on these to 2 parts -- so we'd need to re-do those pieces more efficiently.

**Comment by seananderson** [comment]
Also, you should be able to fit the 3 parts of the ZOIB model as 3 separate models right now and combine the predictions after. The only issues are:
- You won't be able to get uncertainty on combined predictions with the TMB generalized delta method.
- You won't be able to use the `get_index()` function.
- You won't be able to share any parameters across submodels if that's desired.

However, you can get around all of the above (except parameter sharing) by sampling from the posterior with Stan or approximating it with MVN sampling (`nsim` option in predict) and combining predictions with uncertainty that way.

See page 7 in [this paper](https://esajournals.onlinelibrary.wiley.com/doi/full/10.1002/ecs2.2600) where I summarized Liu and Kong (2015) for how you'd combine the 3 models.

**Comment by maxlindmark** [comment]
Thanks for the explanation and the suggestion! Super helpful!
